com.unity.netcode.gameobjects/Prototyping/NetworkTransform.cs

```diff
@@ -152,25 +207,93 @@ public void NetworkSerialize(NetworkSerializer serializer)
         [SerializeField, Range(0, 120), Tooltip("The base amount of sends per seconds to use when range is disabled")]
         public float FixedSendsPerSecond = 30f;
 
+        public bool UseFixedUpdate = true;
+
+        private const int k_DebugDrawLineTime = 10;
+
+        protected virtual IInterpolator<float> PositionXInterpolator { get; set; }
```

Comment:
Haven't thought too much about this but is there a reason why couldn't make all of those into a single interpolator. Or at least just `PositionInterpolator`, `RotationInterpolator` and `ScaleInterpolator` instead of having all the axes separate?

Reply:
I kind of get the "flexibility"/"modularity" idea here but I always think of these things as features.
Do we really need to have super granular, generic, per property interpolators "feature" to succeed? — personally, I don't think so.
These things come with their costs: maintenance, documentation, complexity etc.
I'd suggest bundling position+rotation+scale interpolation into one implementation (let's say `BufferedInterpolator`) and interpolate whatever is appropriate in however way in that implementation, rather than trying to offer "hey we're so generic, flexible and modular!" for no obvious reason, at least at this point in time and development.

Reply:
The separate interpolators are for the "Syncing Position X/Y/Z" options. If a user deselects the X axis for syncing, I wouldn't interpolate that axis anymore, just like I wouldn't sync it anymore.

Reply:
sure, pipe that through NetworkTransform to interpolator implementation so that it'd also know which specific axes it'd interpolate on.

Reply:
would putting buffered interpolator as "internal" make this ok for you? This way we're not exposing a public api we're not ready yet, and we don't have to refactor anything for this release.
We can also potentially reuse this internally in the future for netvar interpolation if we want.